modules/andes-core/broker/src/main/java/org/wso2/andes/server/exchange/topic/TopicExchangeResult.java

```diff
@@ -53,13 +53,15 @@ public void addUnfilteredQueue(AMQQueue queue)
     public void removeUnfilteredQueue(AMQQueue queue)
     {
         Integer instances = _unfilteredQueues.get(queue);
-        if(instances == 1)
-        {
-            _unfilteredQueues.remove(queue);
-        }
-        else
-        {
-            _unfilteredQueues.put(queue,instances - 1);
+
+        // if instances comes up null, that means there was anyway no unfiltered queue instance for the input AMQQueue.
+        // Therefore, only remove if not null.
+        if (null != instances) {
```

Comment:
In the event that an existing subscription without a selector comes back with a new selector, instances can become null.